service/src/main/java/com/commercetools/pspadapter/payone/notification/NotificationProcessorBase.java

```diff
@@ -121,23 +120,20 @@ public final void processTransactionStatusNotification(final Notification notifi
      * the same instance.
      */
     private CompletionStage<Order> updateOrderIfExists(Order order, Payment updatedPayment) {
-        if (order != null) {
-            PaymentState newPaymentState = getPaymentToOrderStateMapper().mapPaymentToOrderState(updatedPayment);
-
-            if (newPaymentState == null && updatedPayment.getPaymentStatus() != null) {
-                logger.warn("Payment [{}] has paymentStatus [{}] which can't be mapped to Order#paymentState. "
-                                + "The order's state remains unchanged.",
-                        updatedPayment.getId(), updatedPayment.getPaymentStatus().getInterfaceCode());
-            }
-
-            // skip update for undefined or unchanged payment state
-            if (newPaymentState != null && !newPaymentState.equals(order.getPaymentState())) {
-                return getOrderService().updateOrderPaymentState(order, newPaymentState);
-            }
-        } else {
-            // This may happen when Payone sends the notification faster then user was redirected back to the shop
-            // and the order has not been created yet.
-            logger.info("Payment [{}] is updated, but respective order is not found!", updatedPayment.getId());
+
+        final PaymentState newPaymentState = getPaymentToOrderStateMapper()
+            .mapPaymentToOrderState(updatedPayment);
+
+        if (newPaymentState == null && updatedPayment.getPaymentStatus() != null) {
+            // TODO do we need this warning?
```

Comment:
@butenkor @ahalberkamp ☝️

Reply:
I will create an issue out of it, so it can be addressed later.

Reply:
https://github.com/commercetools/commercetools-payone-integration/issues/259

Reply:
It's quite nosiy.

Reply:
It seems not necessary but it also means that we do not have a complete mapping here: https://github.com/commercetools/commercetools-payone-integration/blob/5859ac7c8254aa7514f18e491b3d6fee24351941/service/src/main/java/com/commercetools/pspadapter/payone/mapping/order/DefaultPaymentToOrderStateMapper.java#L23
In other words Payone might introduce new status that we can not map yet and thus having wrong status set. We would need to look at some examples to see if this log is required or not.